SingularityUI/app/models/TaskHistoryItem.coffee

```diff
@@ -6,7 +6,19 @@ class TaskHistoryItem extends Model
 
     ignoreAttributes: ['id', 'canBeRunNow']
 
-    initialize: ({ @taskId, @updatedAt, @lastTaskState }) ->
+    url: () ->
+        if @requestId and @runId
+            "#{ config.apiRoot }/history/request/#{ @requestId }/run/#{ @runId }"
+        else
+            # Currently the above URL is the ONLY place to fetch this model.
+            # If you don't have access to request ID and run ID use the TaskHistory
```

Comment:
When you fetch by taskId, you get a different model as a response. That gives back a SingularityTaskHistory (which is the TaskHistory model in the UI), wheras searching by runId returns a SingularityTaskIdHistory (Which is this model in the UI). In a roundabout way, you can use Task Search to search out this model by taskId, but then you'd be fetching the collection rather than the model.

![screenshot 2016-03-31 09 07 55](https://cloud.githubusercontent.com/assets/3210505/14176659/2221720a-f720-11e5-9185-7682fee0e1d8.png)
![screenshot 2016-03-31 09 10 06](https://cloud.githubusercontent.com/assets/3210505/14176714/625af51c-f720-11e5-8eee-738aaaf40579.png)